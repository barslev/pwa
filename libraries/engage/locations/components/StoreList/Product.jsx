import React, { useContext } from 'react';
import { Ellipsis } from '@shopgate/engage/components';
import { ProductImage } from '@shopgate/engage/product';
import FulfillmentContext from '../context';
import * as styles from './style';

/**
 * Renders the product information.
 * @returns {JSX}
 */
function Product() {
  const { product, selectedVariants } = useContext(FulfillmentContext);

  if (!product) {
    return null;
  }

  return (
    <div className={styles.productContainer}>
      <div className={styles.productContainerInner}>
        <div className={styles.productImage}>
          <ProductImage src={product.featuredImageUrl} />
        </div>
        <div className={styles.productContent}>
          <div className={styles.productName}>
            <Ellipsis>{product.name}</Ellipsis>
          </div>
          {selectedVariants && selectedVariants.length > 0 && (
            <div className={styles.variants} data-test-id="selected-variants">
              {selectedVariants.map(({ label, value }) => (
                <div key={`${label}-${value}`}>{`${label}: ${value}`}</div>
              ))}
            </div>
          )}
          <div className={styles.availability}>
            <div className={styles.availabilityText}>Ready for pick up in 1-3 days in</div>
            <div className={styles.availabilityStores}>13 Stores</div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Product;
