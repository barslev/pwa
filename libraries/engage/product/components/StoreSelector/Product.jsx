import React, { useContext } from 'react';
import Ellipsis from '@shopgate/pwa-common/components/Ellipsis';
import ProductImage from '../ProductImage';
import StoreSelectorContext from './context';
import * as styles from './style';

/**
 * Renders the product information.
 * @returns {JSX}
 */
function Product() {
  const { product, selectedVariants } = useContext(StoreSelectorContext);

  if (!product) {
    return null;
  }

  return (
    <div className={styles.productContainer}>
      <div className={styles.productContainerInner}>
        <div className={styles.productImage}>
          <ProductImage src={product.featuredImageUrl} />
        </div>
        <div className={styles.productContent}>
          <div className={styles.productName}>
            <Ellipsis>{product.name}</Ellipsis>
          </div>
          {selectedVariants && selectedVariants.length > 0 && (
            <div className={styles.variants} data-test-id="selected-variants">
              {selectedVariants.map(({ label, value }) => (
                <div key={`${label}-${value}`}>{`${label}: ${value}`}</div>
              ))}
            </div>
          )}
          <div className={styles.availability}>
            <div className={styles.availabilityText}>Ready for pick up in 1-3 days in</div>
            <div className={styles.availabilityStores}>13 Stores</div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Product;
