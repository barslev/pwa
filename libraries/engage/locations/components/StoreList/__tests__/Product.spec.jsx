import React, { useContext } from 'react';
import { shallow } from 'enzyme';
import Product from '../Product';

const product = {
  featuredImageUrl: 'http://www.some.com',
  name: 'Test Product',
};

const selectedVariants = [
  {
    label: 'test',
    value: 'test',
  },
];

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(),
}));

describe('<Product />', () => {
  it('should not render if not product is supplied', () => {
    useContext.mockReturnValueOnce({});
    const wrapper = shallow(<Product context={{}} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render a product without selected variants.', () => {
    useContext.mockReturnValueOnce({ product });
    const wrapper = shallow(<Product context={{}} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Ellipsis').getElements().length).toEqual(1);
    expect(wrapper.find('[data-test-id="selected-variants"]').getElements().length).toEqual(0);
  });

  it('should render a product with selected variants', () => {
    useContext.mockReturnValueOnce({
      product,
      selectedVariants,
    });
    const wrapper = shallow(<Product context={{}} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Ellipsis').getElements().length).toEqual(1);
    expect(wrapper.find('[data-test-id="selected-variants"]').getElements().length).toEqual(1);
  });
});
