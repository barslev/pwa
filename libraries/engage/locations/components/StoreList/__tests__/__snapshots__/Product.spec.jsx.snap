// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> should not render if not product is supplied 1`] = `""`;

exports[`<Product /> should render a product with selected variants 1`] = `
<div
  className={
    Object {
      "data-css-g4i9nv": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1f7wp8g": "",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-19shpyr": "",
        }
      }
    >
      <WithWidgetSettings(ProductImage)
        src="http://www.some.com"
      />
    </div>
    <div
      className={
        Object {
          "data-css-yp5sfk": "",
        }
      }
    >
      <div
        className={
          Object {
            "data-css-9xm049": "",
          }
        }
      >
        <Ellipsis
          className=""
          ellipsis="..."
          rows={3}
        >
          Test Product
        </Ellipsis>
      </div>
      <div
        className={
          Object {
            "data-css-mlrmmn": "",
          }
        }
        data-test-id="selected-variants"
      >
        <div
          key="test-test"
        >
          test: test
        </div>
      </div>
      <div
        className={
          Object {
            "data-css-1f7wp8g": "",
          }
        }
      >
        <div
          className={
            Object {
              "data-css-14jedxl": "",
            }
          }
        >
          Ready for pick up in 1-3 days in
        </div>
        <div
          className={
            Object {
              "data-css-hhr8nm": "",
            }
          }
        >
          13 Stores
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Product /> should render a product without selected variants. 1`] = `
<div
  className={
    Object {
      "data-css-g4i9nv": "",
    }
  }
>
  <div
    className={
      Object {
        "data-css-1f7wp8g": "",
      }
    }
  >
    <div
      className={
        Object {
          "data-css-19shpyr": "",
        }
      }
    >
      <WithWidgetSettings(ProductImage)
        src="http://www.some.com"
      />
    </div>
    <div
      className={
        Object {
          "data-css-yp5sfk": "",
        }
      }
    >
      <div
        className={
          Object {
            "data-css-9xm049": "",
          }
        }
      >
        <Ellipsis
          className=""
          ellipsis="..."
          rows={3}
        >
          Test Product
        </Ellipsis>
      </div>
      <div
        className={
          Object {
            "data-css-1f7wp8g": "",
          }
        }
      >
        <div
          className={
            Object {
              "data-css-14jedxl": "",
            }
          }
        >
          Ready for pick up in 1-3 days in
        </div>
        <div
          className={
            Object {
              "data-css-hhr8nm": "",
            }
          }
        >
          13 Stores
        </div>
      </div>
    </div>
  </div>
</div>
`;
